wasm/src/main/java/com/dylibso/chicory/wasm/Parser.java

```diff
@@ -399,6 +401,19 @@ private static ImportSection parseImportSection(ByteBuffer buffer) {
         return importSection;
     }
 
+    private static MutabilityType getMutabilityType(ByteBuffer buffer) {
+        MutabilityType globalMut = null;
+        try {
+            globalMut = MutabilityType.forId((int) readVarUInt32(buffer));
```

Comment:
For consistency with the "style" started [here](https://github.com/dylibso/chicory/pull/244/files#diff-b617d5b6a9607327caad59874d23af108d8b5dac7d8225f4726db391075c7dd6R141) I would move the throwing of the exception inside the `MutabilityType.forId` method.

cc. @dmlloyd

Reply:
In #249 I've replaced this parsing code with `WasmInputStream#mut`.

Reply:
If I do so I should actually duplicate where the exception is thrown, both here and in `MutabilityType.forId` because in in some cases also the invocation of `readVarUInt32(buffer)` throws an Exception and in this way I'm dealing with both situation here.

Reply:
> in some cases also the invocation of `readVarUInt32(buffer)` throws an Exception

Interesting, do you know in which circumstances?